teps/0050-ignore-task-failures.md

```diff
@@ -0,0 +1,190 @@
+[
+  {
+    "lastTransitionTime": "2021-02-05T18:51:15Z",
+    "message": "Tasks Completed: 1 (Failed: 1, Cancelled 0), Skipped: 3",
+    "reason": "Failed",
+    "status": "False",
+    "type": "Succeeded"
+  }
+]
+```
+
+Tekton [catalog](https://github.com/tektoncd/catalog) has a wide range of `tasks` which are designed to be reusable
+in many pipelines. As a pipeline execution engine, we encourage the pipeline authors to utilize arbitrary tasks from
+the Tekton catalog. But, many common pipelines have the requirement where a task failure must not block executing the
+rest of the tasks.
+
+A pipeline author has an option to utilize `finally` section of the pipeline in which all the final tasks are executed
+after all the tasks in a graph have completed regardless of success or failure. `finally` has its own advantages and
+very helpful in various use cases including notifications, cleanup, etc.
+
+But, the pipeline authors does not have the flexibility to unblock executing the rest of the graph after experiencing a
+single task failure.
+
+
+## Motivation
+
+It should be possible to utilize tasks from the Tekton catalog in a pipeline. A pipeline author has no
+control over the task definitions but may desire to ignore a failure and continue executing the rest of the graph.
+
+
+### Goals
+
+* Design a task failure strategy so that the pipeline author can control the behavior of the underlying tasks 
+  and decide whether to continue executing the rest of the graph in the event of failure.
+
+* Be applicable to any pipeline with references to the tasks in a catalog or inlined task specifications.
+
+### Non-Goals
+
+* Not an alternative to combining the tasks in a pipeline which is covered in
+  [TEP-0044 Composing Tasks with Tasks](https://github.com/tektoncd/community/pull/316).
+
+* Not optimizing pipeline runtime which is covered in
+  [TEP-0046 PipelineRun in a Pod](https://github.com/tektoncd/community/pull/318).
+
+## Requirements
+
+* Users should be able to use any task from the catalog without having to alter its specification to allow that task to
+  fail without stopping the execution of a pipeline.
+
+* It should be possible to know that a task failed, and the rest of the graph was allowed to continue by observing
+  the status of the `PipelineRun`.
+
+* Do not fail the pipeline if any of the tasks fail and continue executing rest of the tasks.
+
+### Use Cases
+
+* As a pipeline author, I would like to design a pipeline where a task running
+  [unit tests](https://github.com/tektoncd/catalog/tree/master/task/golang-test/0.1) might fail,
+  but the pipeline can continue running integration tests and deploying an application to a staging cluster, so that the
+  application can be shared with other developers for early feedback.
+
+  For example, do not fail the pipeline if `unit tests` fail. Continue deploying to a staging cluster if integration tests
```

Comment:
@pritidesai does `do not fail the pipeline` mean that not only does execution continue but also the pipelinerun status should not be failure?

Reply:
or is it just about the execution flow and it doesn't affect the overall status?

Reply:
Hey @jerop the high level idea here is to ignore/allow `pipelineTask` failure, continue execution and do not fail the `pipelineRun`.

> @pritidesai does do not fail the pipeline mean that not only does execution continue but also the pipelinerun status should not be failure?

Yes, `pipelineRun` status should not be failure. I will cover this in the design proposal with potential alternatives. We could set the overall `pipelineRun` status to `success`, `completed`, `success with warning`, or something more relevant.